apis/hive/v1/azure/platform.go

```diff
@@ -24,6 +24,16 @@ type Platform struct {
 	// If empty, the value is equal to "AzurePublicCloud".
 	// +optional
 	CloudName CloudEnvironment `json:"cloudName,omitempty"`
+
+	// ResourceGroupName is the name of an already existing resource group where the cluster should be installed.
+	// This resource group should only be used for this specific cluster and the cluster components will assume
+	// ownership of all resources in the resource group. Destroying the cluster using installer will delete this
+	// resource group.
```

Comment:
@2uasimojo The installer default, when no `resourceGroupName` is provided in the install-config, is to create a resource group named after the cluster `InfraID` and then delete the resource group during uninstall. When `resourceGroupName` is set in install-config, the installer expects that uninstall will delete that resource group so we're now telling uninstall about the resource group when provided but require the `resourceGroupName` to be set within `ClusterDeployment` platform as well as within the install-config for that to function.

I considered moving the field to the MachinePool but, since the cluster was created in this custom resource group, every MachinePool needs to know about it so the ClusterDeployment made sense for a place to configure it more globally. AND we want to honor the installer's docstring re: deleting the custom resource group during uninstall. Also, we are requiring that the user go out of their way to mirror the resource group from install-config to ClusterDeployment platform to make this work which isn't intuitive. 

Azure cluster pools probably wouldn't want to bring a pre-existing resource group ever since something would need to make a new resource group for every cluster pool cluster (because we delete it).

Reply:
It almost makes sense to look inside the install-config (as referenced from the `ClusterDeployment`) for the `resourceGroupName` but we don't do this for any other setting afaik and depending on how implemented could have ramifications on vendoring in the future.

Reply:
...or we could put it in an annotation. Thoughts?